Feature/descw 567 route tests common

mhaswell-bcgov opened the pull request.

mhaswell-bcgov committed:
Created test/routes/index.js to perform common route tests for HTTP status codes 200, 400, 401, 404, 500.
Updated users route tests to use above route tester.
Changed admin_form controller's noQuery() to return a 404 code.
Added missing params property to user validator.

mhaswell-bcgov committed:
Created test/routes/index.js to perform common route tests for HTTP status codes 200, 400, 401, 404, 500.
Updated users route tests to use above route tester.
Changed admin_form controller's noQuery() to return a 404 code.
Added missing params property to user validator.

mhaswell-bcgov committed:
Updated changelog
Changed users controller getAll to respond with 404 when model returns null
Fixed lint issue

mhaswell-bcgov committed:
Updated changelog
Changed users controller getAll to respond with 404 when model returns null
Fixed lint issue

mhaswell-bcgov committed:
Updated changelog

mhaswell-bcgov committed:
Updated changelog

mhaswell-bcgov committed:
Fixed wrong error message in users controller.

mhaswell-bcgov committed:
Fixed wrong error message in users controller.

ASpiteri-BCGov committed:
removed tests, michael is building new ones

mhaswell-bcgov committed:
Removed unnecessary response arguments from users route test and common test suite.

mhaswell-bcgov committed:
Removed unnecessary response arguments from users route test and common test suite.

mhaswell-bcgov commented:
Removed passing in responses as they are unnecessary per @kshapka-bcgov's suggestion.

ASpiteri-BCGov committed:
updated all models

ASpiteri-BCGov committed:
fixed linter errors

ASpiteri-BCGov committed:
added back keycloak test

ShawnTurple commented on backend/test/routes/user.test.js:
```diff
@@ -1,95 +1,54 @@
-    expect(response.statusCode).toBe(401);
-  });
-});
+      payload: {
```
@mhaswell-bcgov this decision was made before you arrived, but try not to include any real names in your tests.

ShawnTurple reviewed (dismissed):
I'm assuming this is just for the users to get feedback, and then to update the rest of the route tests.

ShawnTurple committed:
Merge pull request #191 from bcgov/feature/DESCW-564-Controller-hook

Feature/descw 564 controller hook

mhaswell-bcgov committed:
Merge branch 'feature/DESCW-567-route-tests-common' of github.com:bcgov/gdx-agreements-tracker into feature/DESCW-567-route-tests-common

mhaswell-bcgov committed:
Fixed merge conflict in users controller

mhaswell-bcgov committed:
Fixed user routes test failures caused by changes to capabilities.
Updated routes/index to only test specific routes for 404s.
Removed temporary test/routes/user.js file.
